### DISCUSSION TITLE
How to install in a Windows 10 machine

### DISCUSSION THREAD

**Discussion by hrpitta** [GoogleCloudPlatform/terraformer#1183] [Q&A]
Hey Guys! 
I've been trying to install in a Windows 10 machine (My Company only allows that). Any documentation how to do? I tried to git clone but I'm stuck on that. I want to import from an AWS Environment.
Thanks a lot.
Hugo

**Comment by sunil29feb** [comment]
1. Install Terraform - https://www.terraform.io/downloads
2. Install Terraformer - https://releases.hashicorp.com/terraform-provider-aws/
3. Create a folder and initialize the terraform provider  - refer https://learn.hashicorp.com/tutorials/terraform/aws-build?in=terraform/aws-get-started

**Reply by hrpitta** [reply]
Hey @sunil29feb, I did the step 1 and 3 and they are working perfectly. The step 2 I did the git clone and installed the go lang as well. But I couldn't use the terraformer. terraformer is not recognized as internal or external command.

**Reply by hrpitta** [reply]
Just to inform, I'm not using Chocolatey to install terraformer because I don't have admin access on the computer. Because of that I'm asking the instructions to do manually.

**Reply by sunil29feb** [reply]
@hrpitta  - Please download the exe file (https://github.com/GoogleCloudPlatform/terraformer/releases/download/0.8.18/terraformer-aws-windows-amd64.exe) and use it.

**Reply by hrpitta** [reply]
Thanks @sunil29feb! Was able to import everything. Thanks a lot.

**Reply by hrpitta** [reply]
Another question @sunil29feb. After I generate my .tf files I'm trying to do a terraform plan/apply but I'm receiving an error because Terraform is unable to read the state file. Any clues?

**Reply by sunil29feb** [reply]
@hrpitta - Can you please share the error message?

If you are getting this kind of error -
**Initializing the backend...
╷
│ Error: Invalid legacy provider address**

then run below command-
"terraform state replace-provider -- -/aws hashicorp/aws" against your state file.

**Comment by sunil29feb** [comment]
@hrpitta - did it work for you?

**Reply by hrpitta** [reply]
Hey @sunil29feb, it worked nicely. Having some issues with the Glue Jobs but I'm figuring out. Thank you so much for all the help.

**Comment by hrpitta** [comment]
This is the error! I did the command to import Glue jobs and received that. Do we have the option to import only the jobs? I don't need the crawlers, tables and etc.
![image](https://user-images.githubusercontent.com/44876613/149185724-10ba9d71-b04a-49d9-ad5f-e2001cbdfc5b.png)

**Comment by chetan-alti** [comment]
Hai @sunil29feb , I'm also facing same issue as @hrpitta  when running terraformer.
I have downloaded the [.exe ] (https://github.com/GoogleCloudPlatform/terraformer/releases/download/0.8.24/terraformer-all-windows-amd64) file from release page.
then I have renamed the exe to terraformer and  moved the exe to D://terraformer 
-  I was able to run the terraformer --version (got the result as version v0.8.24)

I have a use-case I'm trying, the following is the prerequiesites
folder structure
Iac_folder
 |_ localmain.tf
 |_ provider.tf

provider.tf
```
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "5.26.0"
    }
  }
}

# Configure the AWS Provider
provider "aws" {
  region = "us-east-1"
  access_key = "acces_key"
  secret_key = "key"
}
```
 localmain.tf
```
terraform {
  backend "local" {
    path = "terraform.tfstate"
  }
}
```

I ran terraform init , it generated `.terrform` folder 

but when I run `terraformer import aws --resources=s3 --regions=us-east-1`
getting following error message
2024/04/19 15:57:45 aws importing region us-east-1
2024/04/19 15:57:45 open \.terraform.d/plugins/windows_amd64: The system cannot find the path specified.

![image](https://github.com/GoogleCloudPlatform/terraformer/assets/118523934/6ca8952b-d831-4577-bd3b-cb921009b31e)


Your help much appreciated , Thanks in advance

**Reply by chetan-alti** [reply]
This is how I resolved the issue, https://github.com/GoogleCloudPlatform/terraformer/issues/1866#issuecomment-2068497606
